packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx

```diff
@@ -132,8 +134,18 @@ export const StyledBaseInput = React.forwardRef<TextInput, StyledBaseInputProps>
         onBlur={(): void => {
           setCurrentInteraction('default');
         }}
-        onChangeText={(text): void => handleOnChange?.({ name, value: text })}
+        onChangeText={(text): void => {
+          handleOnChange?.({ name, value: text });
+          handleOnInput?.({ name, value: text });
+        }}
         onEndEditing={(event): void => handleOnBlur?.({ name, value: event?.nativeEvent.text })}
+        onKeyPress={(event): void => {
+          handleOnKeyDown?.({
+            name,
+            key: event?.nativeEvent.key,
+            event: (event as unknown) as React.KeyboardEvent<HTMLInputElement>, // TODO: handle platform specific type
```

Comment:
Need to fix this to get proper platform-specific types. Will do it while PR is reviewed